src/memray/commands/_attach.lldb

```diff
@@ -15,7 +15,7 @@ breakpoint set -b malloc -b calloc -b realloc -b free -b PyMem_Malloc -b PyMem_C
 # Set commands to execute when breakpoint is reached
 breakpoint command add -e true
 breakpoint disable
-expr auto $dlsym = (void* (*)(void*, const char*))&dlsym
+expr auto $dlsym = (void* (*)(void*, const char*))dlsym.d_original
```

Comment:
This is extremely annoying but seems that `lldb` on Linux resolves `dlsym` to our hook and in macOS it resolves it to the real `dlsym`. I have played wit some alternatives but I cannot make a single script work in both (I also tried fully namespacing it, which fails in macOS because it doesn't resolve `memray`).

@godlygeek Ideas welcomed ;)

Reply:
This fails if `memray` hasn't already been imported:
```
error: Aborting reading of commands after command #1: 'expr auto $dlsym = (void* (*)(void*, const char*))dlsym.d_original'
failed with error: <user expression 11>:1:46: 'dlsym' has unknown type; cast it to its declared type to use it
    1 | auto $dlsym = (void* (*)(void*, const char*))dlsym.d_original
      |                                              ^~~~~
```
The only reason it works for the test suite is that the test program does `from memray._test import MemoryAllocator` which imports `memray._memray` as a side effect, heh. `memray._test` is in the `memray` package, which runs `memray/__init__.py`, which imports `memray._memray`.

Reply:
I _think_ this is happening because we're violating ODR, actually.

We're using `decltype` to declare new functions with the same signature as the functions we're hooking, but `extern "C"` is part of the type, so even though we're putting these things in a namespace, they're mangling to just the original name!

I think the right fix here is namespacing our hooks the C way, by prefixing `memray_` before the original function name.